app/controllers/api/v2/moves_actions.rb

```diff
@@ -0,0 +1,148 @@
+module Api::V2
+  module MovesActions
+    def index_and_render
+      moves = Moves::Finder.new(filter_params, current_ability, params[:sort] || {}).call
+
+      paginate moves,
+               each_serializer: ::V2::MoveSerializer,
+               include: included_relationships,
+               fields: ::V2::MoveSerializer::INCLUDED_FIELDS
+    end
+
+    def show_and_render
+      render_move(move, :ok)
+    end
+
+    def create_and_render
+      move = Move.new(move_attributes)
+      authorize!(:create, move)
+      move.save!
+
+      Notifier.prepare_notifications(topic: move, action_name: 'create')
+
+      render_move(move, :created)
+    end
+
+    def update_and_render
+      raise ActiveRecord::ReadOnlyRecord, "Can't change moves coming from Nomis" if move.from_nomis?
+
+      @move.assign_attributes(move_attributes)
+      action_name = @move.status_changed? ? 'update_status' : 'update'
+      @move.save!
+      @move.allocation&.refresh_status_and_moves_count!
+
+      Notifier.prepare_notifications(topic: @move, action_name: action_name)
+
+      render_move(@move, :ok)
+    end
+
+  private
+
+    PERMITTED_MOVE_PARAMS = [
+      :type,
+      attributes: %i[
+        date
+        time_due
+        status
+        additional_information
+        cancellation_reason
+        cancellation_reason_comment
+        reason_comment
+        move_agreed
+        move_agreed_by
+        move_type
+        date_from
+        date_to
+      ],
+      relationships: {},
+    ].freeze
+
+    def move_params
+      @move_params ||= params.require(:data).permit(PERMITTED_MOVE_PARAMS).to_h
+    end
+
+    def move_attributes
+      move_params[:attributes].tap do |attributes|
+        attributes[:profile] = profile unless profile_attributes.nil?
+        attributes[:from_location] = from_location unless from_location_attributes.nil?
+        attributes[:to_location] = to_location unless to_location_attributes.nil?
+        attributes[:court_hearings] = court_hearings unless court_hearing_attributes.nil?
+        attributes[:prison_transfer_reason] = prison_transfer_reason unless prison_transfer_reason_attributes.nil?
+      end
+    end
+
+    def profile
+      profile_id = profile_attributes.dig(:data, :id)
+
+      Profile.find(profile_id) if profile_id
+    end
+
+    def from_location
+      location_id = from_location_attributes.dig(:data, :id)
+
+      Location.find(location_id) if location_id
+    end
+
+    def to_location
+      location_id = move_params.dig(:relationships, :to_location, :data, :id)
+
+      Location.find(location_id) if location_id
+    end
+
+    def court_hearings
+      court_hearing_ids = court_hearing_attributes.dig(:data)
+      court_hearing_ids = court_hearing_ids&.map { |court_hearing| court_hearing[:id] }
+
+      CourtHearing.where(id: court_hearing_ids) if court_hearing_ids
+    end
```

Comment:
I believe we could even remove the condition, I am gonna test it
```suggestion
    def court_hearings
      court_hearings_data = court_hearing_attributes.dig(:data) || []

      ids = court_hearings_data.map { |court_hearing| court_hearing[:id] }

      CourtHearing.where(id: ids)
    end
```